src/upload-artifact.ts

```diff
@@ -21,7 +21,7 @@ async function run(): Promise<void> {
 
       const artifactClient = create()
       const options: UploadOptions = {
-        continueOnError: true
+        continueOnError: false
```

Comment:
This is a behavior change?

Reply:
I don't think we should silently change behaviors.  If this is really bothering people, consider exposing this option?

Reply:
How did V1 behave? We should keep this consistent and expose an option if users request it

Reply:
Yup, currently if an error is hit and a file can't get uploaded, the problematic files doesn't get uploaded but everything else continues. Users also have to dig into their logs to see that something didn't get uploaded. This will make the step with the upload action fail so users will immediately know that something bad happened during artifact upload. See the comments linked in the issue.

Reply:
Yeah, I understand it's quite annoying that you can't trust the exit code, but this is the behavior since v1 right?  v2 is just being consistent with v1?

Reply:
I think V1 would still fail the step if an upload failed

Reply:
Here is the design decision we came up with [ADR](https://github.com/github/c2c-actions/blob/master/docs/adrs/1123-v2-upload-download-artifact-actions.md#failed-uploads)

> The upload step is failed (which causes the action to fail and turn red)

So this actually makes it consistent with what we planned out earlier. I missed this 😱

The artifact will still be available for download at the end and the reported size will only be that which was uploaded before the failure.

Reply:
> I think V1 would still fail the step if an upload failed

If this is the case, then we are good.  I just read some users mentioned this is consistent with v1.  

We had the window of introducing breaking changes when we add v2.  Now we should be careful about making any breaking changes, even though it was in the design.  We missed the window.  However, if this is the same as v1, I feel better about changing the behavior now.

Reply:
This behavior change only manifests itself if a really bad error is encountered (like the `ECONNRESET` users encountered with 4 retries doing nothing). Ideally there shouldn't be any errors like this :)